### ISSUE TITLE
Source definition deletion silently fails

### ISSUE THREAD

**Issue by monai** [airbytehq/airbyte#9912]
## Environment

- **Airbyte version**:  0.35.12-alpha
- **OS Version / Instance**: macOS
- **Deployment**: Docker
- **Source Connector and version**: any
- **Destination Connector and version**: any
- **Severity**: Low
- **Step where error happened**: Delete source definition

## Current Behavior

API endpoint `/v1/api/v1/source_definitions/delete` deletes all sources created from that definition but doesn't delete the source definition itself.

I traced the issue back to https://github.com/airbytehq/airbyte/pull/8826/files#diff-0cdb545c2c39429b9f43d7df5545a47f2ea6eb600c358733e06973444826d3f5R164 where it deletes all the sources, and this is the last piece of code that works. The later code part doesn't appear to work:

``` java
persistedSourceDefinition.withTombstone(true);
configRepository.writeStandardSourceDefinition(persistedSourceDefinition);
```

If I understand correctly, this code should update config in Airbyte DB table `airbyte_config`, isn't it? However, the table is empty.

## Expected Behavior

I expect API endpoint `/v1/api/v1/source_definitions/delete` to delete the source definition.

## Logs

```
airbyte-server      | 2022-01-31 09:39:39 INFO i.a.s.RequestLogger(filter):95 - REQ 172.18.0.4 POST 204 /api/v1/source_definitions/delete - {"sourceDefinitionId":"dc9a7c70-595f-47c3-aefb-04d562f4a70e"}
airbyte-webapp      | 172.18.0.1 - - [31/Jan/2022:09:39:39 +0000] "POST /api/v1/source_definitions/delete HTTP/1.1" 204 0 "-" "PostmanRuntime/7.29.0" "-"
```

Airbyte DB tables that might be relevant:

`airbyte_config_migrations`:

```json
[
  {
    "installed_rank": 1,
    "version": "0.29.0.001",
    "description": "Baseline from file-based migration v1",
    "type": "BASELINE",
    "script": "Baseline from file-based migration v1",
    "checksum": null,
    "installed_by": "null",
    "installed_on": "2022-01-17 13:09:30.516406",
    "execution_time": 0,
    "success": true
  },
  {
    "installed_rank": 2,
    "version": "0.30.22.001",
    "description": "Store last sync state",
    "type": "JDBC",
    "script": "io.airbyte.db.instance.configs.migrations.V0_30_22_001__Store_last_sync_state",
    "checksum": null,
    "installed_by": "BootloaderApp",
    "installed_on": "2022-01-17 13:09:31.188117",
    "execution_time": 141,
    "success": true
  },
  {
    "installed_rank": 3,
    "version": "0.32.8.001",
    "description": "AirbyteConfigDatabaseDenormalization",
    "type": "JDBC",
    "script": "io.airbyte.db.instance.configs.migrations.V0_32_8_001__AirbyteConfigDatabaseDenormalization",
    "checksum": null,
    "installed_by": "BootloaderApp",
    "installed_on": "2022-01-17 13:09:31.357454",
    "execution_time": 578,
    "success": true
  },
  {
    "installed_rank": 4,
    "version": "0.35.1.001",
    "description": "RemoveForeignKeyFromActorOauth",
    "type": "JDBC",
    "script": "io.airbyte.db.instance.configs.migrations.V0_35_1_001__RemoveForeignKeyFromActorOauth",
    "checksum": null,
    "installed_by": "BootloaderApp",
    "installed_on": "2022-01-17 13:09:31.94899",
    "execution_time": 43,
    "success": true
  }
]
```

`airbyte_metadata`:

```json
[
  {
    "key": "2022-01-17T13:09:32.594194224Z_init_db",
    "value": "0.35.5-alpha"
  },
  {
    "key": "2022-01-18T13:22:00.18007292Z_init_db",
    "value": "0.35.5-alpha"
  },
  {
    "key": "2022-01-25T09:55:49.290151381Z_init_db",
    "value": "0.35.9-alpha"
  },
  {
    "key": "2022-01-31T08:45:50.014303542Z_init_db",
    "value": "0.35.12-alpha"
  },
  {
    "key": "2022-01-31T09:06:54.72477129Z_init_db",
    "value": "0.35.12-alpha"
  },
  {
    "key": "2022-01-31T09:17:28.022268342Z_init_db",
    "value": "0.35.12-alpha"
  },
  {
    "key": "airbyte_version",
    "value": "0.35.12-alpha"
  },
  {
    "key": "deployment_id",
    "value": "bf0d5e2b-e235-4b47-a976-8af38e90e097"
  }
]
```

## Steps to Reproduce

1. Create a custom connector.
2. Call API method `/v1/api/v1/source_definitions/delete` with the connector ID.
3. Check Settings -> Sources

## Are you willing to submit a PR?

No

**Comment by alafanechere** [comment]
According to this [PR](https://github.com/airbytehq/airbyte/pull/8826)  this is something the API support since 0.34.4

@pmossman could you try to reproduce please?

**Comment by pmossman** [comment]
@alafanechere yes I'll look into this today

**Comment by pmossman** [comment]
@alafanechere I believe I found the issue: I added the tombstone field to our source/destination definition JSON schema around the same time a separate project was normalizing database tables ([that PR here](https://github.com/airbytehq/airbyte/pull/8563)). The tombstone field for source/destination definitions was missed when implementing the new config persistence. I'll put a fix together today.

**Comment by pmossman** [comment]
@monai a fix for this was released in version 0.35.17-alpha. The `actor_definition` table now has a `tombstone` column that should be set to `true` when the delete endpoint is called. I'll close this issue but please re-open if your issue persists in the new version

**Comment by monai** [comment]
> @monai a fix for this was released in version 0.35.17-alpha.

Good job! I can confirm that the deletion works now.
